app/views/devise/shared/_links.html.erb

```diff
@@ -1,20 +1,25 @@
-<%- if controller_name != 'sessions' %>
-  <%= link_to "Sign in", new_session_path(resource_name) %><br />
+<%- if controller_name != 'sessions' && controller_name != 'saml_sessions' %>
+  <%- if SiteSetting['MixedSignIn'] || !SiteSetting['SsoSignIn'] %>
```

Comment:
suggestion: This `SiteSetting['MixedSignIn'] || !SiteSetting['SsoSignIn']` seems to be repeated a lot, both here and in other files. Worth creating `sso_sign_in_enabled?` and/or `devise_sign_in_enabled?` helper methods for?

Reply:
Good suggestion, I do wonder how to make the helper method work in the devise emails, since they don't descend from application mailer nor from application controller, and I'm not sure if they load all helpers by default.

Reply:
Have a look in app/views/donations_mailer and admin_mailer - looks like some helpers might be supported out of the box, but if not you can use `ApplicationController.helpers.helper_method_name` to access them.